hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSet.java

```diff
@@ -432,54 +427,27 @@ public long count() {
                                .getExecRowDefinition()
                                .schema());
 
-            //Convert the left operand to a untyped dataset
-            Dataset<Row> right = SpliceSpark.getSession()
-                    .createDataFrame(
-                            ((SparkDataSet)dataSet).rdd
-                                   .map(new LocatedRowToRowFunction()),
-                            context.getOperation()
-                                   .getExecRowDefinition()
-                                   .schema());
-
-            //Do the intesect
-            Dataset<Row> result = left.intersect(right);
-
-            //Convert back to RDD<ExecRow>
-            return toSpliceLocatedRow(result,context);
-
-        }
-        catch (Exception se){
-            throw new RuntimeException(se);
+            return new NativeSparkDataSet(left, context).intersect(dataSet, name, context, pushScope, scopeDetail);
```

Comment:
I'll let SparkDataset handle the union if both sides are SparkDatasets, union is a bit different than intersect since it doesn't require shuffling or anything.

Reply:
Ok, thanks!